corehq/apps/reports/standard/forms/reports.py

```diff
@@ -58,6 +61,30 @@ def submitfilter(self):
             self._submitfilter = SubmissionTypeFilter.get_filter_toggle(self.request)
         return self._submitfilter
 
+    @property
+    def form_filter_params(self):
+        form_filter = FormsByApplicationFilter(self.request, self.domain)
+        params = []
+        for label in form_filter.rendered_labels:
+            param_name = f'{form_filter.slug}_{label[2]}'
+            params.append(dict(
+                name=param_name,
+                value=self.request.GET.get(param_name, None)
+            ))
+        return params
+
+    def _get_app_ids_and_xmlns(self, forms):
+        app_ids = []
+        xmlns_list = []
+        for form in forms:
+            if form['is_fuzzy']:
+                continue
+            if form['app_id']:
+                app_ids.append(form['app_id'])
```

Comment:
If the use has only selected an app then we'll filter by all form submissions for a particular app. If the user selects a specific form then we'll filter to only show submissions for that form. This is where the XMLNS comes into play.

This might make more sense if I explain it with a screenshot:
![form-filter](https://github.com/dimagi/commcare-hq/assets/122617251/6eab4d78-1f13-4623-897f-e2e744f047e2)

If the user only selects an app from (1), then we'll filter for all form submissions for that particular app. If the user also selects (2) then we'll filter to only show form submissions for that module in the selected app. Finally, selecting (3) will cause it to filter submissions only from the specific form. from that app.

Reply:
> Won't all forms have an "app_id"?

That is correct. This is a good point and I think I can actually change the list we use here to a set to make sure we don't have duplicate app ids/xmlns.

> I am unable to connect how what user selects on the UI, is known to this method here to decide where to pick app_id or xmlns.

The params we receive are the following:
- `app_id`
- `module`
- `xmlns`
When we call `self.all_relevant_forms.values()` we get the form definitions for all forms matching the above params. This process for retrieving the relevant form definitions based on our selected filters happens [here](https://github.com/dimagi/commcare-hq/blob/2c9a902fd7d56b5df95ba9eb6879f6ef727695dd/corehq/apps/reports/filters/forms.py#L503) in `_get_selected_forms()`. If, for example, a user only selects (1), then `_get_selected_forms()` will return the form definitions for all forms in that app. We then grab the relevant `app_id` and every form's xmlns from the list of returned form definitions.

Reply:
> If, for example, a user only selects (1), then _get_selected_forms() will return the form definitions for all forms in that app. We then grab the relevant app_id and every form's xmlns from the list of returned form definitions.

So if user selects an app, we take all forms of that app and then take app_id and xmlns from those forms. Okay.

if user selects a form, we take that forms app_id and xmlns. But then we filter forms by app_id, it would give form submissions for all forms of the app and not just the one we selected since we are filtering by app_id?
Did I misunderstand this?

Reply:
> if user selects a form, we take that forms app_id and xmlns. But then we filter forms by app_id, it would give form submissions for all forms of the app and not just the one we selected since we are filtering by app_id?

That is correct. Filtering by a single form means we'll only have one `app_id` and `xmlns` to filter by. Filtering by application (no module or form selected) means we'll have a single `app_id` but possibly many `xmlns`. This would result in showing submissions for all forms in an app.

Reply:
Offlined, thank you @zandre-eng